File: win-prog/ProgWinCS/Beziers and Other Splines/BezierClock/BezierClock.cs
//------------------------------------------
// BezierClock.cs © 2001 by Charles Petzold
//------------------------------------------
using Petzold.ProgrammingWindowsWithCSharp;
using System;
using System.Drawing;
using System.Windows.Forms;

class BezierClock: Form
{
     BezierClockControl clkctl;

     public static void Main()
     {
          Application.Run(new BezierClock());
     }
     public BezierClock()
     {
          Text = "Bezier Clock";

          clkctl = new BezierClockControl();
          clkctl.Parent    = this;
          clkctl.Time      = DateTime.Now;
          clkctl.Dock      = DockStyle.Fill;
          clkctl.BackColor = Color.Black;
          clkctl.ForeColor = Color.White;

          Timer timer = new Timer();
          timer.Interval = 100;
          timer.Tick += new EventHandler(OnTimerTick);
          timer.Start();
     }
     void OnTimerTick(object obj, EventArgs ea)
     {
          clkctl.Time = DateTime.Now;
     }
}

File: win-prog/ProgWinVB/Beziers and Other Splines/BezierClock/BezierClock.vb
'--------------------------------------------
' BezierClock.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class BezierClock
    Inherits Form

    Private clkctrl As BezierClockControl

    Shared Sub Main()
        Application.Run(New BezierClock())
    End Sub

    Sub New()
        Text = "Bezier Clock"

        clkctrl = New BezierClockControl()
        clkctrl.Parent = Me
        clkctrl.Time = DateTime.Now
        clkctrl.Dock = DockStyle.Fill
        clkctrl.BackColor = Color.Black
        clkctrl.ForeColor = Color.White

        Dim tmr As New Timer()
        tmr.Interval = 100
        AddHandler tmr.Tick, AddressOf OnTimerTick
        tmr.Start()
    End Sub

    Private Sub OnTimerTick(ByVal obj As Object, ByVal ea As EventArgs)
        clkctrl.Time = DateTime.Now
    End Sub
End Class
